test/src/index/transport/testMobileSupport.js

```diff
@@ -0,0 +1,201 @@
+    process.removeAllListeners('unhandledRejection');
+
+    this.server = MockServer.init();
+
+    this.server.on('listening', () => {
+      done()
+    });
+  });
+
+  afterEach(function (done) {
+    Transport.prototype.getDriver = this.getDriverBackup;
+    Transport.prototype.createDriverService = this.createDriverServiceBackup;
+
+    CommandGlobals.afterEach.call(this, function () {
+      done();
+    });
+  });
+
+  it('error classes for mobile-web support - RealIosDeviceIdError', function () {
+    let src_folders = [
+      path.join(__dirname, '../../../sampletests/withsubfolders')
+    ];
+
+    return runTests({}, settings({
+      output: false,
+      src_folders,
+      desiredCapabilities: {
+        browserName: 'safari',
+        platformName: 'iOS',
+        alwaysMatch: {
+          acceptInsecureCerts: false
+        }
+      },
+
+      webdriver: {
+        start_process: true,
+        server_path: '',
+        cli_args: [
+          // --verbose
+        ]
+      }
+    })).catch(err => {
+      assert.ok(err instanceof Error);
+      assert.strictEqual(err.name, 'RealIosDeviceIdError');
+      assert.strictEqual(err.message, 'Real Device ID is neither configured nor passed');
+      assert.strictEqual(err.help.length, 4)
+    })
+  });
+
+  it('error classes for mobile-web support - IosSessionNotCreatedError', function () {
+    Transport.prototype.getDriver = function() {
+      const err = new Error('An error occurred while creating a new SafariDriver session');
+      err.name = 'SessionNotCreatedError';
+
+      throw err;
+    };
+
+    Transport.prototype.createDriverService = async function() { 
+      this.driverService = {
+        getOutputFilePath(){},
+        getSettingsFormatted(){}
+      };
+    };
+
+    let src_folders = [
+      path.join(__dirname, '../../../sampletests/withfailures'),
+      path.join(__dirname, '../../../sampletests/withsubfolders')
+    ];
+
+    let globals = {
+      calls: 0,
+      retryAssertionTimeout: 0,
+      reporter(results, cb) {
+        assert.ok(results.lastError instanceof Error);
+        assert.ok(results.lastError instanceof IosSessionNotCreatedError);
+        assert.ok(Object.prototype.hasOwnProperty.call(results.lastError, 'name'));
+        assert.ok(Object.prototype.hasOwnProperty.call(results.lastError, 'message'));
+        assert.ok(results.lastError.help.length, 3);
+        cb();
+      }
+    };
+
+    return runTests({
+    }, settings({
+      output: false,
+      src_folders,
+      globals,
+      desiredCapabilities: {
+        browserName: 'safari',
+        platformName: 'iOS',
+        'safari:deviceUDID': '00008030-00024C2C3453402E',
+        alwaysMatch: {
+          acceptInsecureCerts: false
+        }
+      },
+    
+      webdriver: {
+        start_process: true,
+        server_path: '',
+        cli_args: [
+          // --verbose
+        ]
+      }
+    }))
+  });
+
+  it('deviceId passed as argument', function () {
```

Comment:
> Also, these tests are passing even if we revert the changes made in this PR, while they should have failed there. So, these are again not making sure that the deviceId will be added to the desired capabilities in the actual test runs.

Test Fixed

> Since the deviceId tests are now defined here, can we remove the similar tests present in testCliRunner.js, as those are just duplicate?

Duplicacy removed